src/irmin-unix/xgit.ml

```diff
@@ -14,13 +14,19 @@
  * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  *)
 
+open Lwt.Syntax
 include Xgit_intf
 
 let src = Logs.Src.create "git.unix" ~doc:"logs git's unix events"
 
 module Log = (val Logs.src_log src : Logs.LOG)
 
-let remote ?(ctx = Mimic.empty) ?headers uri =
+let remote ?ctx ?headers uri =
+  let+ ctx =
+    match ctx with
+    | Some x -> Lwt.return x
+    | None -> Git_unix.ctx (Happy_eyeballs_lwt.create ())
```

Comment:
I would like to say that the `'a Lwt.t` is not really needed on my side: I did such layout to be homogeneous with `git-mirage` & `functoria` where the `connect` function (which returns the `Mimic.ctx`) should be wrapped inside an `'a Lwt.t`. If it's too noisy for you to append the `'a Lwt.t`, I can try to propose something at the next release of `git-unix`.

Then, the PR seems fine.

Reply:
I don't have an issue with the Lwt change in this PR, but if anyone else does I'm happy to help work out how to change it for the next release of `irmin-git` like you said.